src/scripts/install_machine.py

```diff
@@ -0,0 +1,324 @@
+        params = [
+            "virt-install",
+            "--connect", f"qemu:///{args['connectionName']}",
+            "--quiet",
+            "--name", args['vmName'],
+            "--os-variant", args['os'],
+            "--memory", str(args['memorySize']),
+        ]
+
+        if args['type'] == 'install' or ('storagePool' in args and args['storagePool'] != 'NewVolume'):
+            params += ["--check", "path_in_use=off"]
+
+        if args['sourceType'] != 'disk_image':
+            params += ["--wait", "-1"]
+
+        if args['type'] == 'install' or args['startVm']:
+            params.append("--noautoconsole")
+
+        # Disks
+        if args['type'] == 'install':
+            disks = args['disks'] or ['none']
+            for disk in disks:
+                params += ['--disk', disk]
+        else:
+            params.append("--disk")
+            if args['sourceType'] == "cloud":
+                backing_store = f",backing_store={args['source']},"
+            else:
+                backing_store = ""
+
+            if args['sourceType'] == 'disk_image':
+                params.append(f"{args['source']},device=disk")
+            elif args['storagePool'] == 'NoStorage':
+                params.append("none")
+            elif args['storagePool'] != 'NewVolume':
+                params.append(f"vol={args['storagePool']}/{args['storageVolume']}{backing_store}")
+            else:
+                params.append(f"size={args['storageSize']},format=qcow2{backing_store}")
+
+        # NICs
+        if args['type'] == "install":
+            vnics = args['vnics'] or ['none']
+            for network in vnics:
+                params += ['--network', network]
+
+        # Consoles
+        if args['type'] == "install":
+            graphics = args['graphics'] or ['none']
+            for console in graphics:
+                params += ['--graphics', console]
+        else:
+            has_param = False
+            for graphics in get_graphics_capabilies(args['connectionName']):
+                params += ['--graphics', f"{graphics},listen=127.0.0.1"]
+                has_param = True
+
+            if not has_param:
+                params += ['--graphics', "none"]
+
+        # Installation media
+        if args['sourceType'] != "cloud":
+            prepare_installation_source(params, args)
+
+        # Autostart
+        if args.get('autostart'):
+            params.append("--autostart")
+
+        # VCPUs
+        if 'vcpu' in args:
+            params += ['--vcpus', args['vcpu']]
+
+        # Firmware
+        if 'firmware' in args:
+            params += ['--boot', args['firmware']]
+
+        params += unattended_params
+        params += cloud_init_params
+
+        logging.debug(params)
+
+        yield params
+
+
+def create_vm(args):
+    with prepare_virt_install_params(args) as params:
+        xml = subprocess.check_output(params)
+
+        if args['startVm']:
```

Comment:
I guess everything starting here can be unintended to be outside of the install params context: you only need the temporary files to survive until the subprocess call returns, right?

Reply:
Indeed, just virt-install needs the files.